app/controllers/api/functions.php

```diff
@@ -486,6 +503,82 @@
 
         $eventsInstance->setParam('functionId', $function->getId());
 
+        // activate the deployment for first run of a VCS repo
+        if ($needToDeploy) {
+            $deploymentId = ID::unique();
+            $entrypoint = 'index.js'; //TODO: Read from function settings
+            $deployment = $dbForProject->getDocument('deployments', $deploymentId);
+            $privateKey = App::getEnv('VCS_GITHUB_PRIVATE_KEY');
+            $githubAppId = App::getEnv('VCS_GITHUB_APP_ID');
+            $projectInternalId = $project->getInternalId();
+
+            $vcs_installations = $dbForConsole->findOne('vcs_installations', [
+                Query::equal('projectInternalId', [$projectInternalId])
+              ]);
+            $installationId = $vcs_installations->getAttribute('installationId');
```

Comment:
Why do we need to do this? We can use `$function->getAttribute('installationId')`.

Also this would break I think, because we do `findOne` and there can be multiple installations for one project.

Reply:
Removed this section of code as discussed.